### PR TITLE
feat: Added support for MDC

### PR DESCRIPTION
**Pull request by karelrochelt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,9 +38,69 @@ debug('I am debug')
...
+##### MDC.put(key, value)
+Sets the `value` of `key` in MDC.
+
+##### MDC.remove(key)
+Sets the `key` to `undefined`.
+
+##### MDC.clear()
+Removes all data from MDC.
+
+##### One example to explain them all
```
Comment (zatziky) [inline README.md:98]: Cha cha :D

File: README.md [hunk 2]
```diff
@@ -38,9 +38,69 @@ debug('I am debug')
...
+```
+MDC.createContext(() => {
+  MDC.put('foo', '1')
+  doStuff(() => {
```
Comment (zatziky) [inline README.md]: This doesn't make much sense to me. `doStuff(cb)` is provided a function but when is this function gonna be executed?

File: src/request-scope-storage.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const continuationStorage = require('continuation-local-storage')
```
Comment (zatziky) [inline src/request-scope-storage.js]: Shall we rename the file to `mdc.js` or `mapped-diagnostic-context.js`? You call it MDC from other files anyways...

**Review by zatziky — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -38,9 +38,69 @@ debug('I am debug')
...
+
+#### MDC API
+##### MDC.createContext(next)
+Creates MDC context. Has to be called before calling `put`, otherwise `put` will have no effect.
+
+It is possible to create nested contexts. In that case, all MDC data from parent context will be copied to child context, therefore all modifications to MDC in the child context will be done independently from the parent context.
+
+The `next` parameter is context callback. MDC data will be valid during the whole duration of the callback. This makes it ideal to call this in a middleware function (see [request-id.middleware.js](src/request-id.middleware.js) for example).
+```
+const MDC = require('logzio-node-debug').MDC
```
Comment (zatziky) [inline README.md:54]: I would provide an example with a middleware function. I myself cannot imagine the usage.

File: src/request-scope-storage.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  const mdcCopy = Object.assign({}, mdc)
+  requestNamespace.run(() => {
+    requestNamespace.set(KEY_MDC, mdcCopy)
+    next()
+  })
+}
+
+function getAll() {
+  const mdc = getNamespace(NSP_REQUEST).get(KEY_MDC)
+  if(!mdc) {
```
Comment (zatziky) [inline src/request-scope-storage.js]:
```
  if(!mdc) { return {} }
  
  return mdc
```

can be rewritten as `return mdc || {}`

File: README.md (hunk 2 above)
Comment (karelrochelt) [reply README.md]: Yeah, this didn't make sense at all 😄

File: README.md (hunk 4 above)
Comment (karelrochelt) [reply README.md:54]: Added

File: README.md [hunk 6]
```diff
@@ -86,16 +99,18 @@ Removes all data from MDC.
...
-    MDC.getAll() // => {foo: '1', bar: 2}
-    MDC.remove('foo')
-    MDC.getAll() // => {bar: 2}
-    MDC.clear()
-    MDC.getAll() // => {}
-  })
+  doStuff()
 })
+
+function doStuff() {
```
Comment (zatziky) [inline README.md]: tuhle funkci můžeš inlajnovat. :D

**Review by zatziky — APPROVED** [review]

**Pull request merged**